Sources/MapboxMaps/Style/Sources.swift

```diff
@@ -30,12 +27,32 @@ public enum SourceType: String, Codable {
         case rasterDem = "raster-dem"
         case geojson = "geojson"
         case image = "image"
-        case video = "video"
         case model = "model"
     }
+
+    /// The associated Swift struct type
+    public var sourceType: Source.Type {
+        switch self {
+        case .vector:
+            return VectorSource.self
+        case .raster:
+            return RasterSource.self
+        case .rasterDem:
+            return RasterDemSource.self
+        case .geoJson:
+            return GeoJSONSource.self
+        case .image:
+            return ImageSource.self
+        case .model:
+            return ModelSource.self
+        }
+    }
 }
 
-public protocol Source: Codable, StyleEncodable, StyleDecodable { }
+public protocol Source: Codable, StyleEncodable, StyleDecodable {
+    /// Rendering type of this source.
+    var type: SourceType { get }
+}
```

Comment:
Anything else we want to add to this protocol? Doing so post-GA would be a breaking change.